### DISCUSSION TITLE
C++ intellisence Problem

### DISCUSSION THREAD

**Discussion by kushuaiming** [microsoft/vscode-cpptools#12540] [Q&A]
My Environment is Ubuntu 20.04, VSCode Version 1.92.0
Almost all build in c++ library cannot find kernel file, and the error log is like this: cannot open source file "asm/unistd_nr_n32.h"
My c_cpp_propablities.json file is:
![image](https://github.com/user-attachments/assets/330cf090-3a5f-498e-9c2f-2cca0da47d9a)
I have install kernel files by use `sudo apt-get install linux-headers-$(uname -r)` and this solve some "cannot find files" problem, but there are still some files cannot find, such as  "asm/unistd_nr_n32.h".
How can i download or install something, so c++ intellisence can find "asm/unistd_nr_n32.h", or how can i modify settings so the c++ intellisence no need to find this file?
![image](https://github.com/user-attachments/assets/491300d6-6904-4336-97c7-5451690442e1)

**Comment by bobbrow** [comment]
Hi @kushuaiming what is the path to `asm/unistd_nr_n32.h` on your computer?

**Reply by kushuaiming** [reply]
I use find command like this, but cannot find `asm/unistd_nr_n32.h`
`cd /`
`sudo find . -name unistd_nr_n32.h`

**Reply by bobbrow** [reply]
Unfortunately, I don't know what packages you need to install to get this header on your machine.  But it looks like the header is nowhere to be found in your filesystem, so that is why there is an error reported by IntelliSense.
